services/src/datasets/add_from_directory.rs

```diff
@@ -74,29 +79,48 @@ pub async fn add_providers_from_directory<D: LayerProviderDb>(db: &mut D, file_p
         Ok(())
     }
 
-    let dir = fs::read_dir(file_path);
-    if dir.is_err() {
-        warn!("Skipped adding providers from directory because it can't be read");
-        return;
-    }
-    let dir = dir.expect("checked");
+    async fn add_from_dir<D: LayerProviderDb>(db: &mut D, file_path: &Path) {
```

Comment:
```suggestion
    async fn add_providers_from_directory<D: LayerProviderDb>(db: &mut D, file_path: &Path) {
```

maybe?

The other comments apply here as well about internals